hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -281,18 +278,25 @@ private Diagnostics initDiagnostics() {
         return new Diagnostics(name, logger, instanceName, properties);
     }
 
-    private MetricsRegistryImpl initMetricsRegistry() {
-        ProbeLevel probeLevel = properties.getEnum(Diagnostics.METRICS_LEVEL, ProbeLevel.class);
-        ILogger logger = loggingService.getLogger(MetricsRegistryImpl.class);
-        MetricsRegistryImpl metricsRegistry = new MetricsRegistryImpl(getName(), logger, probeLevel);
-        RuntimeMetricSet.register(metricsRegistry);
-        GarbageCollectionMetricSet.register(metricsRegistry);
-        OperatingSystemMetricSet.register(metricsRegistry);
-        ThreadMetricSet.register(metricsRegistry);
-        ClassLoadingMetricSet.register(metricsRegistry);
-        FileMetricSet.register(metricsRegistry);
-        metricsRegistry.scanAndRegister(clientExtension.getMemoryStats(), "memory");
-        return metricsRegistry;
+    private void initProbeSources() {
+        probeRegistry.register(GcProbeSource.INSTANCE);
+        probeRegistry.register(MachineProbeSource.INSTANCE);
+        probeRegistry.register(this);
+        probeRegistry.register(statistics);
+        NioNetworking networking = connectionManager.getNetworking();
+        if (networking != null) {
+            // null when testing with mock
+            probeRegistry.register(networking);
+        }
+        probeRegistry.registerIfSource(clientExtension);
+    }
+
+    @Override
```

Comment:
Static objects should be registered explicitly

So 

```
probeRegistry.register(invocationService);
probeRegistry.register(executionService);
probeRegistry.register(listenerService);
probeRegistry.register(clientExtension.getMemoryStats())
```

Now it is spread over multiple methods. The additional registrations are not a concern regarding memory since the number is limited and even if a single source has n probes, it doesn't lead to n registration, but just one.

Reply:
Static objects are now registered or get collected as they are annotated with `@ProbeSource`.